### ISSUE TITLE
How to auto create/export/produce annotation files for objects yolo detects on image?

### ISSUE THREAD

**Issue by easy-and-simple** [AlexeyAB/darknet#6240]
Hello
I want to train my model to detect few classes, lets say, cats, dogs, humans but I am not happy with images in dataset I have, so I need to find images that I need and to make annotations for them in pascal voc format.
I don't want to use some tool to create annotations manually, so I need to use yolo to detect objects on images I supply and in addition of writing object boxes to write and annotation file for every image with data for all objects.
I searched for way how to set yolo to create/export annotation files when detection objects but didn't found info if this is possible and if it is what I need to do to set yolo to do the job

So how to set yolo to create/export/produce annotation files in pascal voc and other formats for objects detected in images I have, and how to set yolo to process all images in given folder?

If there is no such option could you implement it, because it is very easy to implement and very useful and needed

**Comment by biparnakroy** [comment]
i did a similar project on this have a look : https://github.com/biparnakroy/pseudoLabelGeneratorForYOLO

**Comment by AlexeyAB** [comment]
https://github.com/AlexeyAB/darknet#how-to-use-on-the-command-line

Pseudo-lableing - to process a list of images data/new_train.txt and save results of detection in Yolo training format for each image as label <image_name>.txt (in this way you can increase the amount of training data) use: `./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights -thresh 0.25 -dont_show -save_labels < data/new_train.txt`

**Comment by easy-and-simple** [comment]
Hello guys,
Thank you for your replays!

@AlexeyAB
It is not working. I get following error:
Cannot load image "-dont_show"
STB Reason: can't fopen

**Comment by sa7ina** [comment]
After you generate labels with yolo you the classes have their numbers with the pretrained class number.
If you want to remap them to your custom class numbers i wrote a simple remapping tool for that a few days ago:
[https://github.com/sa7ina/YOLOv2-5_Class_Remap](url)

**Comment by biparnakroy** [comment]
@easy-and-simple  even i faced the same issue while using ```-save_results``` and ```-dont_show```

**Comment by biparnakroy** [comment]
ok so here's the thing the model outputs the coordinates of the top left corner of the bounding box, whereas the labels need the coordinates of the center of the bounding box, that might be the problem, so u need to convert those

**Comment by AlexeyAB** [comment]
It works well:

![image](https://user-images.githubusercontent.com/4096485/87682315-5e4ff700-c788-11ea-99f9-acddf29fec0b.png)


![image](https://user-images.githubusercontent.com/4096485/87682336-63ad4180-c788-11ea-9a8e-06732d734dbd.png)

![image](https://user-images.githubusercontent.com/4096485/87682489-8f302c00-c788-11ea-80ea-8c242657bfde.png)

**Comment by easy-and-simple** [comment]
@AlexeyAB 
So in order to work, path to images needs to be absolute right?
Also this functionality is not enough for my needs because I can't see how correct are bounding boxes and labels. I expect good accuracy but I must be sure that all is OK and I am not wasting time to train my model on wrong data.

@biparnakroy 
I needed some very quick and easy solution, that I can use without effort, but I probably will need to find time to make some copy/paste programming and to make something that will do the job as I need it. Also values that it writes to these files are obviously not in pixels

**Comment by biparnakroy** [comment]
@easy-and-simple yolo follows this format : https://github.com/AlexeyAB/Yolo_mark/issues/60#issuecomment-401854885
beside u can use this https://github.com/biparnakroy/pseudoLabelGeneratorForYOLO its basically what you wanted

**Comment by AlexeyAB** [comment]
> So in order to work, path to images needs to be absolute right?

I use relative paths, content of my `images.txt` file:
```
image1.jpg
image2.jpg
image3.jpg
```
